UANodeManager.cs

```diff
@@ -174,95 +186,107 @@ public override void CreateAddressSpace(IDictionary<NodeId, IList<IReference>> e
                 {
                     try
                     {
-                        AddAsset(references, file, out ThingDescription td, out FolderState assetFolder);
+                        var assetId = Path.GetFileNameWithoutExtension(file);
+                        Log.Logger.Information($"Configuring asset: {assetId}");
 
+                        ParseAsset(file, out ThingDescription td);
                         AddOPCUACompanionSpecNodes(td);
-
+                        AddAsset(references, td, out BaseObjectState assetFolder, assetId);
+                        
                         // create nodes for each TD property
                         foreach (KeyValuePair<string, Property> property in td.Properties)
                         {
                             foreach (object form in property.Value.Forms)
                             {
                                 if (td.Base.ToLower().StartsWith("modbus://"))
                                 {
-                                    AddModbusNodes(td, assetFolder, property, form);
+                                    AddModbusNodes(td, assetFolder, property, form, assetId);
                                 }
                             }
                         }
 
                         AddPredefinedNode(SystemContext, assetFolder);
 
-                        _ = Task.Factory.StartNew(UpdateNodeValues, td.Title + " [" + td.Name + "]", TaskCreationOptions.LongRunning);
+                        _ = Task.Factory.StartNew(UpdateNodeValues, assetId/*td.Title + " [" + td.Name + "]"*/, TaskCreationOptions.LongRunning);

```

Comment:
Remove, don't comment out.

Reply:
Seems to have been left there by mistake, I'll remove it.